.ci/build_demo.sh

```diff
@@ -0,0 +1,143 @@
+#!/bin/bash -e
+
+# Builds a new demo instance using Docker Compose.
+# Uses URLs <demo_name>.loki.shopsys.com and 2.<demo_name>.loki.shopsys.com
+# Adds nginx configuration into /etc/nginx/conf.d/<demo_name>.conf
+#
+# Usage: build_demo.sh <demo_name>
+#
+# Example: build_demo.sh my-demo
+
+# ANSI color codes
+RED="\e[31m"
+GREEN="\e[32m"
+BLUE="\e[34m"
+NC="\e[0m"
+
+DEMO_NAME=$1
+JOB_NAME=${JOB_NAME:-$DEMO_NAME}
+WORKSPACE=${WORKSPACE:-$PWD}
+ELASTICSEARCH_NETWORK=${ELASTICSEARCH_NETWORK:-shared-demo-elasticsearch-network}
+ELASTICSEARCH_CONTAINER=${ELASTICSEARCH_CONTAINER:-shared-demo-elasticsearch-instance}
+
+if [[ ! "$DEMO_NAME" =~ ^[a-z0-9]([a-z0-9-]*[a-z0-9])?$ ]]; then
+    echo -e "${RED}Invalid demo name!${NC}"
+    echo -e "${BLUE}Use only lowercase letters and numbers, you can include dashes in the middle (eg. \"my-demo-42\").${NC}"
+    exit 1
+fi
+
+echo -e "${BLUE}Configuring your demo-shop...${NC}"
+
+# Configuration files are baked into the Docker image at the moment
+cp "$WORKSPACE/app/config/parameters.yml.dist" "$WORKSPACE/app/config/parameters.yml"
+cp "$WORKSPACE/app/config/domains_urls.yml.dist" "$WORKSPACE/app/config/domains_urls.yml"
+
+# Disable master e-mail and mailer whitelist
+sed -i "s/mailer_master_email_address:.*/mailer_master_email_address: ~/" "$WORKSPACE/app/config/parameters.yml"
+sed -i "s/mailer_delivery_whitelist:.*/mailer_delivery_whitelist: ~/" "$WORKSPACE/app/config/parameters.yml"
+
+# Fetch all domain IDs
+DOMAIN_IDS=$(cat "$WORKSPACE/app/config/domains_urls.yml" | grep -Po 'id: ([0-9]+)$' | sed 's/id: \([0-9]\+\)/\1/')
+
+# Modify public URLs to $DOMAIN_ID.$DEMO_NAME.loki.shopsys.com ($DOMAIN_ID is omitted for the first domain)
+for DOMAIN_ID in $DOMAIN_IDS; do
+    if [[ "$DOMAIN_ID" == "1" ]]; then
+        # 1st domain has URL without a number prefix
+        sed -i "/id: 1/,/url:/{s/url:.*/url: http:\/\/$DEMO_NAME.loki.shopsys.com/}" "$WORKSPACE/app/config/domains_urls.yml"
+    else
+        # 2nd and subsequent domains have URLs with DOMAIN_ID prefix
+        sed -i "/id: $DOMAIN_ID/,/url:/{s/url:.*/url: http:\/\/$DOMAIN_ID.$DEMO_NAME.loki.shopsys.com/}" "$WORKSPACE/app/config/domains_urls.yml"
+    fi
+done
+
+echo -e "${BLUE}Connecting to shared elasticsearch instance...${NC}"
+
+NETWORK_CONNECT='false'
+
+if docker network create "$ELASTICSEARCH_NETWORK" &> /dev/null; then
+    echo "A network $ELASTICSEARCH_NETWORK has been created."
+    NETWORK_CONNECT='true'
+else
+    echo "The network $ELASTICSEARCH_NETWORK already exists."
+fi
+
+if [[ -z "$(docker ps -q -f name="$ELASTICSEARCH_CONTAINER")" ]]; then
```

Comment:
this would match also `$ELASTICSEARCH_CONTAINER-blablablablabla` container but i think this is low probability.

Reply:
I was aware of that and came to the same conclusion :slightly_smiling_face: :crossed_fingers: